contracts/modules/cover/ProductsV1.sol

```diff
@@ -0,0 +1,1383 @@
+    //"supportedChains": ["ethereum"],
+    //"type": "token",
+    //"symbol": "yvDAI",
+    //"coveredToken": "0x19D3364A399d251E894aC732651be8B0E4e85001",
+    //"underlyingToken": "DAI",
+    //"dateAdded": "2021-05-25",
+    //"logo": "yearn.svg"
+    if (legacyProductId == 0x0000000000000000000000000000000000000006)  {
+      return 0;
+    }
+
+    //"name": "Yearn yvUSDC v2",
+    //"site": "https://yearn.finance",
+    //"supportedChains": ["ethereum"],
+    //"type": "token",
+    //"symbol": "yvUSDC",
+    //"coveredToken": "0x5f18C75AbDAe578b483E5F43f12a39cF75b973a9",
+    //"underlyingToken": "DAI",
+    //"dateAdded": "2021-05-25",
+    //"logo": "yearn.svg"
+    if (legacyProductId == 0x0000000000000000000000000000000000000007)  {
+      return 0;
+    }
+
+    //"name": "Yearn ycrvstETH v2",
+    //"site": "https://yearn.finance",
+    //"supportedChains": ["ethereum"],
+    //"type": "token",
+    //"symbol": "ycrvstETH",
+    //"coveredToken": "0xdCD90C7f6324cfa40d7169ef80b12031770B4325",
+    //"underlyingToken": "ETH",
+    //"dateAdded": "2021-05-25",
+    //"logo": "yearn.svg"
+    if (legacyProductId == 0x0000000000000000000000000000000000000008)  {
+      return 0;
+    }
+
+    //"name": "Curve 3pool LP (3CRV)",
+    //"site": "https://curve.fi/",
+    //"supportedChains": ["ethereum"],
+    //"type": "token",
+    //"symbol": "3CRV",
+    //"coveredToken": "0x6c3f90f043a72fa612cbac8115ee7e52bde6e490",
+    //"underlyingToken": "DAI",
+    //"dateAdded": "2021-05-25",
+    //"logo": "curve-3crv.png"
+    if (legacyProductId == 0x0000000000000000000000000000000000000009)  {
+      return 0;
+    }
+
+
+    //"name": "Curve sETH LP (eCRV)",
+    //"site": "https://curve.fi/",
+    //"supportedChains": ["ethereum"],
+    //"type": "token",
+    //"symbol": "eCRV",
+    //"coveredToken": "0xA3D87FffcE63B53E0d54fAa1cc983B7eB0b74A9c",
+    //"underlyingToken": "ETH",
+    //"dateAdded": "2021-05-25",
+    //"logo": "curve.png"
+    if (legacyProductId == 0x0000000000000000000000000000000000000010)  {
+      return 0;
+    }
+
+    //"name": "Idle DAI v4 (idleDAIYield)",
+    //"site": "https://idle.finance/",
+    //"supportedChains": ["ethereum"],
+    //"type": "token",
+    //"symbol": "idleDAIYield",
+    //"coveredToken": "0x3fE7940616e5Bc47b0775a0dccf6237893353bB4",
+    //"underlyingToken": "DAI",
+    //"dateAdded": "2021-07-05",
+    //"logo": "idleDAIYield.svg"
+    if (legacyProductId == 0x0000000000000000000000000000000000000011)  {
+      return 0;
+    }
+
+    //"name": "Idle USDT v4 (idleUSDTYield)",
+    //"site": "https://idle.finance/",
+    //"supportedChains": ["ethereum"],
+    //"type": "token",
+    //"symbol": "idleUSDTYield",
+    //"coveredToken": "0xF34842d05A1c888Ca02769A633DF37177415C2f8",
+    //"underlyingToken": "DAI",
+    //"dateAdded": "2021-07-05",
+    //"logo": "idleUSDTYield.svg"
+    if (legacyProductId == 0x0000000000000000000000000000000000000012)  {
```

Comment:
curious what's the gas cost of this last branching path (worst case)

Reply:
26866 gas units for the last one and 22116 gas units for the first one. Subtracting the base gas fee and adding the cost of a call of 2600 it means that the gas cost increase is 8466 and 3716 respectively.

Reply:
ok so the gas costs would be lower if we just do SLOAD, but the relative cost increase to the base cost of the contract call is not that high